TestCNIPlugin flakes

**Which jobs are failing**:

pull-kubernetes-bazel-test

**Which test(s) are failing**:

//pkg/kubelet/dockershim/network/cni:go_default_test

```
--- FAIL: TestCNIPlugin (0.11s)
    cni_test.go:293: failed to parse reported cni input config /tmp/cni-test548751309/output/test_vendor887.in: (<nil> invalid character 'w' after top-level value)
    cni_test.go:300: mismatch in expected port mappings. expected [map[containerPort:80 hostIP:0.0.0.0 hostPort:8008 protocol:udp]] got []
    cni_test.go:307: mismatch in expected bandwidth. expected map[egressBurst:2.147483647e+09 egressRate:1e+06 ingressBurst:2.147483647e+09 ingressRate:1e+06] got map[]
    cni_test.go:317: mismatch in expected ipRange. expected [[map[subnet:10.0.2.0/24]]] got []
    cni_test.go:341: PASS:	AddNetworkList(*context.emptyCtx,*libcni.NetworkConfigList,mock.AnythingOfTypeArgument)
    cni_test.go:341: PASS:	DelNetworkList(*context.emptyCtx,*libcni.NetworkConfigList,mock.AnythingOfTypeArgument)
```

**Since when has it been failing**:

Unsure, it doesn't appear in triage, but I see it regularly in PRs.

**Testgrid link**:

This package is the top flake currently
https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-bazel-test&sort-by-flakiness=

**Reason for failure**:

**Anything else we need to know**:

/sig node
/sig network

--- a/pkg/kubelet/dockershim/network/cni/cni_test.go
+++ b/pkg/kubelet/dockershim/network/cni/cni_test.go
@@ -20,7 +20,6 @@ package cni
 
 import (
 	"bytes"
-	"context"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
@@ -76,14 +75,18 @@ func installPluginUnderTest(t *testing.T, testBinDir, testConfDir, testDataDir, 
 	f, err = os.Create(pluginExec)
 	require.NoError(t, err)
 
+	// TODO: use mock instead of fake shell script plugin
 	const execScriptTempl = `#!/usr/bin/env bash
+echo -n "{ \"cniVersion\": \"{{.CNIVersion}}\", \"ip4\": { \"ip\": \"{{.PodIP}}/24\" } }"
+if [ "$CNI_COMMAND" = "VERSION" ]; then
+	exit
+fi
 cat > {{.InputFile}}
 env > {{.OutputEnv}}
 echo "%@" >> {{.OutputEnv}}
 export $(echo ${CNI_ARGS} | sed 's/;/ /g') &> /dev/null
 mkdir -p {{.OutputDir}} &> /dev/null
-echo -n "$CNI_COMMAND $CNI_NETNS $K8S_POD_NAMESPACE $K8S_POD_NAME $K8S_POD_INFRA_CONTAINER_ID" >& {{.OutputFile}}
-echo -n "{ \"cniVersion\": \"{{.CNIVersion}}\", \"ip4\": { \"ip\": \"{{.PodIP}}/24\" } }"`
+echo -n "$CNI_COMMAND $CNI_NETNS $K8S_POD_NAMESPACE $K8S_POD_NAME $K8S_POD_INFRA_CONTAINER_ID" >& {{.OutputFile}}`
 
 	inputFile := path.Join(testDataDir, binName+".in")
 	outputFile := path.Join(testDataDir, binName+".out")
@@ -225,8 +228,8 @@ func TestCNIPlugin(t *testing.T) {
 	cniPlugin.execer = fexec
 	cniPlugin.loNetwork.CNIConfig = mockLoCNI
 
-	mockLoCNI.On("AddNetworkList", context.TODO(), cniPlugin.loNetwork.NetworkConfig, mock.AnythingOfType("*libcni.RuntimeConf")).Return(&types020.Result{IP4: &types020.IPConfig{IP: net.IPNet{IP: []byte{127, 0, 0, 1}}}}, nil)
-	mockLoCNI.On("DelNetworkList", context.TODO(), cniPlugin.loNetwork.NetworkConfig, mock.AnythingOfType("*libcni.RuntimeConf")).Return(nil)
+	mockLoCNI.On("AddNetworkList", mock.AnythingOfType("*context.emptyCtx"), cniPlugin.loNetwork.NetworkConfig, mock.AnythingOfType("*libcni.RuntimeConf")).Return(&types020.Result{IP4: &types020.IPConfig{IP: net.IPNet{IP: []byte{127, 0, 0, 1}}}}, nil)
+	mockLoCNI.On("DelNetworkList", mock.AnythingOfType("*context.emptyCtx"), cniPlugin.loNetwork.NetworkConfig, mock.AnythingOfType("*libcni.RuntimeConf")).Return(nil)
 
 	// Check that status returns an error
 	if err := cniPlugin.Status(); err == nil {

